packages/scoutgame/src/users/mergeUserAccount.ts

```diff
@@ -0,0 +1,215 @@
+        updatedUserData.displayName = mergedUser.displayName;
+        updatedUserData.bio = mergedUser.bio;
+        updatedUserData.email = mergedUser.email;
+        updatedUserData.walletENS = mergedUser.walletENS;
+      }
+
+      // Detach the identities from the merged user
+      await tx.scout.update({
+        where: {
+          id: mergedUserId
+        },
+        data: {
+          email: null,
+          farcasterId: farcasterId ? null : undefined,
+          farcasterName: farcasterId ? null : undefined,
+          telegramId: telegramId ? null : undefined
+        }
+      });
+
+      await tx.scout.update({
+        where: { id: retainedUserId },
+        data: updatedUserData
+      });
+
+      await prisma.scoutWallet.updateMany({
+        where: {
+          scoutId: mergedUserId
+        },
+        data: {
+          scoutId: retainedUserId
+        }
+      });
+
+      await tx.scout.update({
+        where: { id: mergedUserId },
+        data: {
+          deletedAt: new Date()
+        }
+      });
+
+      await tx.scoutMergeEvent.create({
+        data: {
+          mergedFromId: mergedUserId,
+          mergedToId: retainedUserId,
+          mergedRecords: {
+            farcasterId,
+            telegramId
+          }
+        }
+      });
+
+      await tx.nFTPurchaseEvent.updateMany({
+        where: {
+          scoutId: mergedUserId
+        },
+        data: {
+          scoutId: retainedUserId
+        }
+      });
+
+      await tx.pointsReceipt.updateMany({
+        where: {
+          recipientId: mergedUserId
+        },
+        data: {
+          recipientId: retainedUserId
+        }
+      });
+
+      await tx.pointsReceipt.updateMany({
+        where: {
+          senderId: mergedUserId
+        },
+        data: {
+          senderId: retainedUserId
+        }
+      });
+
+      await tx.tokensReceipt.updateMany({
+        where: {
+          recipientId: mergedUserId
+        },
+        data: {
+          recipientId: retainedUserId
+        }
+      });
+
+      await tx.tokensReceipt.updateMany({
+        where: {
+          senderId: mergedUserId
+        },
+        data: {
+          senderId: retainedUserId
+        }
+      });
+
+      await tx.pendingNftTransaction.updateMany({
+        where: {
+          userId: mergedUserId
+        },
+        data: {
+          userId: retainedUserId
+        }
+      });
+
+      await tx.referralCodeEvent.updateMany({
+        where: {
+          refereeId: mergedUserId
+        },
+        data: {
+          refereeId: retainedUserId
+        }
+      });
+
+      await tx.talentProfile.updateMany({
+        where: {
+          builderId: mergedUserId
+        },
+        data: {
+          builderId: retainedUserId
+        }
+      });
+
+      await tx.scoutGameActivity.updateMany({
+        where: {
+          userId: mergedUserId
+        },
+        data: {
+          userId: retainedUserId
+        }
+      });
```

Comment:
also builderEvent probably? Just in case.. i think we are using it now for non-builder things sometimes.

Reply:
Also: ScoutSocialQuest, ScoutDailyClaimEvent, and ScoutDailyClaimStreakEvent

Reply:
builder event is not necessary as we will always keep the builder intact, and there are unique validation on the rest of the records to not add it for those. eg partner rewards, scout social etc.